### ISSUE TITLE
Planning attempts parameter ignored?

### ISSUE THREAD

**Issue by bit-pirate** [moveit/moveit_ros#359]
Although I'm setting the number of planning attempts higher then 1 in my planning request, ...

```
motion_plan_request.num_planning_attempts = 3
```

... this parameter seems to be ignored when planning.

```
[ INFO] [1383559809.615582701]: Planning attempt 1 of at most 1
```

Any idea why?

**Comment by isucan** [comment]
Probably a bug... Are you using the action or the service interface?

**Comment by bit-pirate** [comment]
> Probably a bug... Are you using the action or the service interface?

The MoveGroup action.

**Comment by sachinchitta** [comment]
There are two different parameters that control different types of behaviors 

(1) there's num_planning_attempts in the motion planning request that controls the (total) number of threads that a planner will create to solve your problem - e.g. ompl will create that many threads and then try to "hybridize" the plans together. Setting num_planning_attempts to 3 (as you are doing) only controls this behavior. 

(2) There's a pair of fields in PlanningOptions (inside the MoveGroupGoal) for "replanning" - this controls the behavior of the entire system during planning and execution. The ROS_INFO you are seeing corresponds to this part of the code - you are probably not setting these fields and so they default. The flag replan controls whether move_group will attempt to replan when a trajectory (during execution) gets invalidated for some reason. The replan_attempts number controls how many such attempts are made to replan before move_group gives up and declares failure. 

In summary, if you want to control (with ompl) the number of planning threads, set num_planning_attempts. If you want to control the number of times that replanning happens during execution, set replan and replan_attempts. 

I think the names/comments on those parameters may be confusing and we should probably fix them but this behavior itself is not a bug.

**Comment by bit-pirate** [comment]
Thanks, Sachin, for clarifying this! I was already aware of the replanning option and think that the description of that parameter is sufficient (see http://docs.ros.org/hydro/api/moveit_msgs/html/msg/PlanningOptions.html). For the planning attempts extra information about the threading would be nice (http://docs.ros.org/hydro/api/moveit_msgs/html/msg/MotionPlanRequest.html). I'll open an extra issue for this.

Following up your description of the parallel (right?) planning, are the parallel executed plannings equivalent to serial (re-)planning? So, there is no difference between executing the planner with one thread three times and executing it one time with three threads?

**Comment by isucan** [comment]
Thanks for clarifying Sachin! The message the bug starts with is actually from the replanning component, and there is just one, so you see 1 planning attempt. num_planning_attempts is not actually the number of threads used for planning though, it is the number of times the OMPL planner is executed to compute a solution, as long as there is time left. All of these solutions are then shortened and ran through a path hybridization algorithm (as they are found) and the shortest solution found is reported. There is an internal parameter to OMPL that sets the maximum number of threads to use (set to 4), so you will get groups of 4 plans computed at a time. The hybridization also happens in parallel, as solutions are found (http://ompl.kavrakilab.org/classompl_1_1tools_1_1ParallelPlan.html is used)
